src/PKSim.Core/Services/PKAnalysesTask.cs

```diff
@@ -251,5 +258,45 @@ private void updateBodyWeightFromCurrentIndividual(IParameter bodyWeightParamete
 
          bodyWeightParameter.Value = allBodyWeights.Count > individualId ? allBodyWeights[individualId] : double.NaN;
       }
+
+      public PKAnalysis CreatePKAnalysisFromValues(PKValues pkValues, Simulation simulation, Compound compound)
+      {
+         var options = _pkCalculationOptionsFactory.CreateFor(simulation, compound.Name);
+         return _pkMapper.MapFrom(compound.MolWeight, pkValues, options.PKParameterMode, compound.Name);
+      }
+
+      private Regex _rangeRegex => new Regex(@"^(.*)Range (\d*)% to (\d*)%");
```

Comment:
This is a field. Move it at the top
Also why do you instantiate it all the time? It can be a readonly field, no?